spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/log/Slf4jScopeDecorator.java

```diff
@@ -43,196 +41,48 @@
-				log("Closing scope for span: {}", currentSpan);
-				scope.close();
-				replace("traceId", previousTraceId);
-				replace("parentId", previousParentId);
-				replace("spanId", previousSpanId);
-				replace("spanExportable", spanExportable);
-				replace(LEGACY_TRACE_ID_NAME, legacyPreviousTraceId);
-				replace(LEGACY_PARENT_ID_NAME, legacyPreviousParentId);
-				replace(LEGACY_SPAN_ID_NAME, legacyPreviousSpanId);
-				replace(LEGACY_EXPORTABLE_NAME, legacySpanExportable);
-				for (AbstractMap.SimpleEntry<String, String> entry : previousMdc) {
-					replace(entry.getKey(), entry.getValue());
-				}
-			}
-
-		}
-		return new ThreadContextCurrentTraceContextScope();
+		this.delegate = builder.build();
 	}
 
-	private List<AbstractMap.SimpleEntry<String, String>> previousMdc() {
-		List<AbstractMap.SimpleEntry<String, String>> previousMdc = new ArrayList<>();
-		List<String> keys = new ArrayList<>(whitelistedBaggageKeys());
-		keys.addAll(whitelistedPropagationKeys());
-		keys.addAll(whitelistedLocalKeys());
-		for (String key : keys) {
-			previousMdc.add(new AbstractMap.SimpleEntry<>(key, MDC.get(key)));
-		}
-		return previousMdc;
-	}
-
-	private List<String> whitelistedKeys(List<String> keysToFilter) {
-		List<String> keys = new ArrayList<>();
-		for (String baggageKey : keysToFilter) {
-			if (this.sleuthSlf4jProperties.getWhitelistedMdcKeys().contains(baggageKey)) {
-				keys.add(baggageKey);
-			}
-		}
-		return keys;
-	}
-
-	private List<String> whitelistedBaggageKeys() {
-		return whitelistedKeys(this.sleuthProperties.getBaggageKeys());
-	}
-
-	private List<String> whitelistedKeysWithValue(TraceContext context,
-			List<String> keys) {
-		if (context == null) {
-			return Collections.EMPTY_LIST;
-		}
-		List<String> nonEmpty = new ArrayList<>();
-		for (String key : keys) {
-			if (StringUtils.hasText(ExtraFieldPropagation.get(context, key))) {
-				nonEmpty.add(key);
-			}
-		}
-		return nonEmpty;
-	}
-
-	private List<String> whitelistedBaggageKeysWithValue(TraceContext context) {
-		return whitelistedKeysWithValue(context, whitelistedBaggageKeys());
-	}
-
-	private List<String> whitelistedPropagationKeys() {
-		return whitelistedKeys(this.sleuthProperties.getPropagationKeys());
-	}
-
-	private List<String> whitelistedLocalKeys() {
-		return whitelistedKeys(this.sleuthProperties.getLocalKeys());
-	}
-
-	private List<String> whitelistedPropagationKeysWithValue(TraceContext context) {
-		return whitelistedKeysWithValue(context, whitelistedPropagationKeys());
-	}
-
-	private List<String> whitelistedLocalKeysWithValue(TraceContext context) {
-		return whitelistedKeysWithValue(context, whitelistedLocalKeys());
-	}
-
-	private void log(String text, TraceContext span) {
-		if (span == null) {
-			return;
-		}
-		if (log.isTraceEnabled()) {
-			log.trace(text, span);
-		}
+	@Override
+	public Scope decorateScope(TraceContext context, Scope scope) {
+		return LEGACY_IDS.decorateScope(context, delegate.decorateScope(context, scope));
```

Comment:
this is effectively the same as mapping dual names to the span identifiers. From a design POV, brave chose to only allow one mapped name per field-decorator. this reduces complexity and improves speed.  Most users will not need to sync multiple names to the same trace context field, but those who can, may use multiple decorators. Typically, we'd wire each decorator as a separate `@Bean` this trick is made just because this change is to rewrite the internals, not break up the type itself.